### PR TITLE
(fix) Ensure form fields are initialised with all base validators

### PR DESCRIPTION
**Pull request by samuelmale**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This aims at fixing the issue where field validation wasn't happening as expected on "value change"

## Related Issue

https://openmrs.atlassian.net/browse/O3-3208

### CODE REVIEW

**Review by denniskigen — COMMENTED** [review]
Mostly LGTM, @samuelmale. Thanks. 

However, I'm concerned that I can't click or tab away from an invalid field. What are you doing to trap focus? It should be possible to click or tab away from an invalid field.

https://github.com/openmrs/openmrs-form-engine-lib/assets/8509731/dce6ff1a-1181-49a1-a5fa-1b88a26130bf

**Comment by samuelmale** [comment]
@denniskigen I understand the issue and plan to fix it. The changes of this PR just unmasked the issue by enabling the validator. However, it's fair to have it fixed as part of this work.

File: __mocks__/forms/afe-forms/test-schema-transformer-form.json [hunk 1]
```diff
@@ -19,6 +23,7 @@
               "label": "Numeric",
               "id": "dem_numeric",
               "type": "obs",
+              "required": "false",
```
Comment (pirupius) [inline __mocks__/forms/afe-forms/test-schema-transformer-form.json:26]: Curious about the need for this. Isn't this by default `false`?

File: src/components/encounter/encounter-form.component.tsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ import type {
 import FormPage from '../page/form-page.component';
 import { FormContext } from '../../form-context';
 import {
-  cascadeVisibityToChildFields,
+  cascadeVisibilityToChildFields,
```
Comment (pirupius) [inline src/components/encounter/encounter-form.component.tsx:16]: Nice catch on the typo 🦾

File: src/transformers/default-schema-transformer.test.ts [hunk 3]
```diff
@@ -1,59 +1,115 @@
-import { AngularFormEngineSchemaTransformer } from './angular-fe-schema-transformer';
-import testForm from '__mocks__/forms/afe-forms/afe-schema-trasformer-form.json';
+import { DefaultFormSchemaTransformer } from './default-schema-transformer';
+import testForm from '__mocks__/forms/afe-forms/test-schema-transformer-form.json';
 
 const expectedTransformedSchema = {
   name: 'AFE form with aliased questions',
+  readonly: false,
```
Comment (pirupius) [inline src/transformers/default-schema-transformer.test.ts:6]: Pardon me if I might have missed this in previous PR's, will the transformer add this key automatically? and  is this in anyway related to the mode?

**Review by pirupius — COMMENTED** [review]
I just have some minor questions regarding some schema changes

File: __mocks__/forms/afe-forms/test-schema-transformer-form.json (hunk 1 above)
Comment (samuelmale) [reply __mocks__/forms/afe-forms/test-schema-transformer-form.json:26]: This is to test out that after the schema transformation we end up with a boolean

File: src/transformers/default-schema-transformer.test.ts (hunk 3 above)
Comment (samuelmale) [reply src/transformers/default-schema-transformer.test.ts:6]: If the property is present, the transformer will attempt to parse to a boolean. For more context, see [thread](https://openmrs.slack.com/archives/C04QZ5DDVMG/p1715788770872919?thread_ts=1715769639.667939&cid=C04QZ5DDVMG).

**Review by pirupius — APPROVED** [review]

**Comment by samuelmale** [comment]
@denniskigen the video below demonstrates the behaviour with the latest changes:

https://www.loom.com/share/71c180ea52c440b58b338547bffbae6b?sid=d05d54f0-0111-44c6-9206-9ca1c7efb0f0

**Comment by samuelmale** [comment]
Nice catch @arodidev! Working on a fix.

File: src/components/inputs/text/text.component.tsx [hunk 4]
```diff
@@ -27,11 +28,12 @@ const TextField: React.FC<FormFieldProps> = ({ question, onChange, handler, prev
     }
   }, [previousValue]);
 
-  field.onBlur = () => {
+  field.onBlur = (event) => {
```
Comment (arodidev) [inline src/components/inputs/text/text.component.tsx:31]: Do we need this `event` here?

**Review by arodidev — APPROVED** [review]
Thanks @samuelmale LGTM, left a minor review comment.

**Comment by arodidev** [comment]
Merged due to time-sensitive nature of work done in this PR.

**Pull request merged**